library/std/src/sys/unix/time.rs

```diff
@@ -293,7 +294,13 @@ mod inner {
 
     impl Instant {
         pub fn now() -> Instant {
-            Instant { t: now(libc::CLOCK_MONOTONIC) }
+            let t = match now(libc::CLOCK_BOOTTIME) {
```

Comment:
I think this should be Linux-specific, and other platforms should just use `CLOCK_MONOTONIC` the first time.

Reply:
Actually, this should be Linux-specific *and* only on x86 or x86-64, nowhere else. Every other target requires a sufficiently recent kernel to guarantee support for `CLOCK_BOOTTIME`.

Reply:
We can only use a constant that our libc actually defines anyways, or it will break compilation for that platform.

Reply:
Please please don't limit it to `x86 and `x86_64`, that defeats the much of the purpose of using `CLOCK_BOOTTIME` - since the difference is for cases where the device goes into suspend, which are mostly `arm` and `arm64` devices. Limiting it to Linux seems expected.

Reply:
@maurer Clarification: the *fallback* should be limited to x86 and x86_64. We should use `CLOCK_BOOTTIME` on every Linux target. Only `x86` and `x86_64` still support sufficiently old kernels that `CLOCK_BOOTTIME` may not exist, so we only need the fallback to `CLOCK_MONOTONIC` on those targets.

Reply:
OK, I am in agreement there, thanks for the clarification.

Reply:
@joshtriplett powerpc, powerpc64, and s390x also document 2.6.32 support.

Reply:
@cuviper You're entirely right, and I don't know how I missed those.